<launch>

  <arg name="rviz" default="true" />

  <group if="$(arg rviz)">
    <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find sl_sensor_visualise)/rviz/structured_light_odometry.rviz" />

    <node pkg="sl_sensor_visualise" type="pose_stamped_to_path_node" name="pose_stamped_to_path_node" output="screen">
      <param name="subscribe_topic" type="str" value="registered_pose" />
      <param name="publish_topic" type="str" value="path" />
    </node>
  </group>

  <node pkg="tf" type="static_transform_publisher" name="camera_init_to_map"  args="0 0 0 0 0 0 1 map my_frame 10" />

  <arg name="node_start_delay" default="5.0" />  

  <?ignore?>
  <!-- ROS Subscribers -->

  <param name="/use_sim_time" value="true"/>

  <node pkg="sl_sensor_vo" type="structured_light_odometry_node" name="structured_light_odometry_node" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' "/>
  <param name="display_correspondences" type="bool" value= "false" />  
  <param name="publish_pose_stamped" type="bool" value= "true" />  
  <param name="publish_correspondences" type="bool" value= "true" />  
  <param name="yaml_directory" type="str" value= "/home/ltf/catkin_ws/src/sl_sensor/core/sl_sensor_vo/odometry_yaml/smb.yaml" />
  <param name="calibration_filename" type="str" value= "/home/ltf/ma_thesis_submission_dataset/vo_demo/calibration.xml"/>
  <param name="scaling_factor" type="double" value="1e-3"/>

  <!-- Input Type Related Params -->
  <param name="input_type" type="int" value="1" />
  <param name="projector_trigger_topic" type="str" value= "/versavis/projector/image_time" />  
  <param name="image_topic" type="str" value= "/versavis/cam1/image_raw" />  
  <param name="number_patterns_per_frame" type="int" value= "5" />  
  <param name="image_tolerance_time" type="double" value= "0.015" />  
  <param name="time_per_pattern" type="double" value= "0.03333333"/>  

  <?ignore?>
  <node pkg="rosbag" type="play" name="player" args="--clock --rate=1 /home/ltf/ma_thesis_submission_dataset/vo_demo/rumlang1.bag" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' "/>
  <!-- Uncomment below if you want csv file logs of the relative and absolute translation outputs-->
  <!--param name="output_relative_pose_csv_filename" type="str" value="/home/ltf/ma_thesis_submission_dataset/vo_demo/vo_relative_poses.csv"/-->
  <!--param name="output_frame_timings_csv_filename" type="str" value="/home/ltf/ma_thesis_submission_dataset/vo_demo/vo_frame_timings.csv"/-->

</launch>
